### ISSUE TITLE
I used the whisper-small model that was downloaded on AI Hub, but its output was not correct.

### ISSUE THREAD

**Issue by MRRRKING** [quic/ai-hub-models#263]
I modified the [app.py](https://github.com/quic/ai-hub-models/blob/main/qai_hub_models/models/_shared/hf_whisper/app.py#L90) file to generate the decoder inputs, and saved the inputs locally. Next, I ran qnn shell to get token ids.  Finally, I compared the results and found that they were not equal.

The whisper-small model is downloaded at https://huggingface.co/qualcomm/Whisper-Small/tree/v0.39.0/precompiled/qualcomm-qcs8550-proxy.

### python

```
python -m qai_hub_models.models.whisper_small.demo --audio-file '/home/jincan/Downloads/audio_files/135_mono.wav'
```

The output is as below. 
It produced 40 tokens, and I compared the second result(50359) whose index is 1.

```
tokens:
[50258, 50260, 50359, 50363, 11100, 6135, 50157, 15003, 13545, 32981, 4275, 10123, 41520, 255, 11189, 11100, 27080, 37390, 21001, 12744, 31706, 2289, 1654, 12467, 19497, 8350, 5155, 2412, 10440, 13876, 23786, 6723, 27908, 31706, 1960, 6734, 10123, 41520, 255, 14769, 50257]

Transcription: 明天晚上我们公司要吃饭但是明晚周五放假了我打算回家有什么理由可以请假不去吃饭吗
```

### qnn(version: 2.38.0.250901)

```
qnn-net-run --retrieve_context ./decoder_model_htp.bin --backend ./libQnnHtp.so --input_list ./decoder_input_list.txt --output_dir ./decoder_output/for01/
```

```
>>> import numpy as np
>>> a = np.fromfile('./logits01.raw', dtype=np.float32)
>>> np.argmax(a)
50260
```

The qnn output(**50260**) and the python output(**50359**) are inconsistent. 
What's the difference between the python env and the qnn env?

The modified app.py is as below:
[app.py](https://github.com/user-attachments/files/23234601/app.py)

**Comment by MRRRKING** [comment]
Any suggestions? Thanks.

**Comment by kory** [comment]
HI @MRRRKING,

The SDK QAIRT may not have bit-exact reimplementations of ops running on your hardware through PyTorch. Some small output variance is expected compared to the CPU implementation, but this should not meaningfully change results.

We have tested end to end accuracy (LibriSpeech) with our whisper model. Accuracy on device is within .15% of floating point accuracy via PyTorch.

I suggest trying our on-device demo & evaluation scripts [here](https://github.com/quic/ai-hub-apps/tree/main/apps/windows/python/Whisper) to try Whisper end-to-end via the QAIRT SDK.
